server/api/src/test/scala/com/prisma/api/mutations/OptionalBackrelationSpec.scala

```diff
@@ -0,0 +1,179 @@
+package com.prisma.api.mutations
+
+import com.prisma.api.ApiBaseSpec
+import com.prisma.shared.models.Project
+import com.prisma.shared.schema_dsl.SchemaDsl
+import org.scalatest.{FlatSpec, Matchers}
+
+class OptionalBackrelationSpec extends FlatSpec with Matchers with ApiBaseSpec {
+
+  "Nested Update" should "be generated for models with missing backrelations " in {
+
+    val project = SchemaDsl.fromString() {
+      """
+        |type Owner {
+        |  id: ID!
+        |  ownerName: String! @unique
+        |  cat: Cat
+        |}
+        |
+        |type Cat {
+        |  id: ID!
+        |  catName: String! @unique
+        |}
+        |
+      """.stripMargin
+    }
+    database.setup(project)
+
+    createItem(project, "Cat", "garfield")
+    createItem(project, "Owner", "jon")
+
+    //set initial owner
+    val res = server.executeQuerySimple(
+      """mutation {updateOwner(where: {ownerName: "jon"},
+        |data: {cat: {connect: {catName: "garfield"}}}) {
+        |    ownerName
+        |    cat {
+        |      catName
+        |    }
+        |  }
+        |}""".stripMargin,
+      project
+    )
+
+    val res2 = server.executeQuerySimple(
+      """mutation {updateOwner(where: {ownerName: "jon"},
+        |data: {cat: {update: { where:{catName: "garfield"}, data: {catName: "azrael"}}}}) {
+        |    ownerName
+        |    cat {
+        |      catName
+        |    }
+        |  }
+        |}""".stripMargin,
+      project
+    )
```

Comment:
why isn't the result checked here? 🤔

Reply:
I only wanted to check that the mutation was now working and was relying on the must fail check.